### DISCUSSION TITLE
[Dev Tools] Foundryup zksync install script

### DISCUSSION THREAD

**Discussion by gretzke** [zkSync-Community-Hub/zksync-developers#696] [Dev Tooling] [answered]
### Environment

Mainnet

### Select the Dev Tool you are using

Foundry

### Provide the version of the tool (if applicable)

latest

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

I’m running into issues with the installation script halfway through.
```
Downloading the install script...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2265  100  2265    0     0  23740      0 --:--:-- --:--:-- --:--:-- 23842
Making install script executable...
Running the installation script...
Installing foundryup-zksync...

Detected your preferred shell is zsh and added foundryup-zksync to PATH.
Run 'source /Users/daniel/.zshenv' or start a new terminal session to use foundryup-zksync.
Then, simply run 'foundryup-zksync' to install foundry-zksync.
Sourcing the shell configuration file: '/Users/daniel/.zshenv'
Downloading foundryup-zksync...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0Warning: Failed to open the file foundryup-zksync: Is a directory
  0  9945    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (56) Failure writing output to destination, passed 821 returned 4294967295
```

### Additional Details

Also the documentation needs a small adjustment as the name of the installer file changed.

**Comment by alhenaai** [comment]
It seems like you are encountering an issue with the installation script for `foundryup-zksync`. The error message indicates a problem with writing the output to the destination, which might be due to incorrect permissions or a misconfiguration in the path settings.

Here are a few steps you can try to resolve this issue:

1. **Check Directory Permissions:**
   Ensure that you have the necessary permissions to write to the directory where `foundryup-zksync` is being downloaded. You can adjust the permissions using the `chmod` command.

2. **Verify Path Settings:**
   Make sure that the path settings in your `.zshenv` are correctly configured to include the directory where `foundryup-zksync` is located. You might need to edit the `.zshenv` file to correct the path.

3. **Re-run the Installation Script:**
   After making the necessary adjustments, try running the installation script again. Ensure that you are executing the script with sufficient privileges (using `sudo` if necessary).

Regarding the documentation adjustment for the installer file name, thank you for pointing that out. It would be beneficial to update the documentation to reflect the correct name of the installer file to avoid confusion.

If these steps do not resolve the issue, it might be helpful to look into the specific configurations of your system or consider downloading the file manually if the script continues to fail.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by dutterbutter** [comment]
Hey @gretzke 👋 

What machine are you using? And are you running this command locally? 

```
curl -L https://raw.githubusercontent.com/matter-labs/foundry-zksync/main/install-foundry-zksync | bash
```

**Reply by gretzke** [reply]
I'm using an M1 Mac. I'm using official instructions from the documentation but the command you sent me also fails

**Reply by dutterbutter** [reply, marked as answer]
@gretzke have you already cloned the repo? Specifically, are you running the command in the `foundry-zksync` directory? If so, please do so from elsewhere. You can see a similar issue we had on our CI due to this: https://github.com/matter-labs/foundry-zksync/pull/573

We have updates to the foundry-book installation section incoming: https://github.com/matter-labs/foundry-zksync-book/pull/17. Apologies on the the inconvenience.

**Reply by gretzke** [reply]
executing the script from the parent directory worked, thanks!

**Reply by dutterbutter** [reply]
Glad it worked! @gretzke for uniswap v4 migration please be aware of the [reserved address range](https://foundry-book.zksync.io/zksync-specifics/limitations/general#reserved-address-range), and changes needed for `.code` usage as described [here](https://foundry-book.zksync.io/zksync-specifics/limitations/compilation#contract-bytecode-access). 

If you experience any pain points along the way please let us know as we are actively documenting / fixing to improve `foundry-zksync` for our community and builders :)

**Reply by gretzke** [reply]
oof, the `.code` limitation is pretty bad because the library we use for gas metering relies on it for example. We are only using `.code` in unit tests afaik but skipping the compilation of tests using the `--skip` flag doesn't seem to be working.

```
Compiler run failed:
Error
Error: LLVM IR generator definition pass: 4762:17 The `EXTCODECOPY` instruction is not supported
--> test/CustomAccounting.t.sol:CustomAccountingTest

Error
Error: LLVM IR generator definition pass: 4788:17 The `EXTCODECOPY` instruction is not supported
--> test/DynamicFees.t.sol:TestDynamicFees

...
```

**Reply by dutterbutter** [reply]
@gretzke You can skip compilation of specific contracts by using `--avoid-contracts` flag. Example usage:
```
forge build --zksync --avoid-contracts "**/CustomAccounting.t.sol,**/DynamicFees.t.sol,**/PoolManager.t.sol"
```

**Reply by 0xhung295** [reply]
@gretzke I'm having an similar issue:
![image](https://github.com/user-attachments/assets/c8d6a1a0-f403-4c24-8035-0b06552c3900)
How can I solve in this case?
